test/syscall/syscall.py

```diff
@@ -195,6 +195,27 @@ def __setitem__(self, key, value):
 pwrite_in = r'pwrite64'
 pwrite_out = r'pwrite'
 
+# Known environments issues and messages with suggestions on how to fix them
+env_known_issues = {
+    'Darwin' : {
+        "dtrace: system integrity protection is on, some features will not be available" :
+            "Test skipped due to the inability to trace system calls since macOS SIP is enabled."
+    }
+}
+
+# Return codes
+class TestReturnCode(enum.Enum):
```

Comment:
I like this change, it showcases the ability to return multiple different return codes. I am slightly worried that this is an anti-pattern of how we treat return codes for our testing. In the WT test codebase, we usually just return -1 and follow the output log rather than focus on the error code.

Reply:
I thought about this too, but as I understand, we do already have some logic that relies on return codes, eg:

https://github.com/wiredtiger/wiredtiger/blob/16dd2186ce9e2382c0edaa7e8aa448b818136fda/test/suite/run.py#L334

vs 

https://github.com/wiredtiger/wiredtiger/blob/16dd2186ce9e2382c0edaa7e8aa448b818136fda/test/suite/run.py#L623

That's why I decided to create a `BAD_INPUT` error code and set it to 2 since I had already observed that this particular error code is used to report bad usage. 

Also, it is worth noting that Cmake has functionality that allows skipping test basing on regular expressions (https://cmake.org/cmake/help/latest/prop_test/SKIP_REGULAR_EXPRESSION.html#prop_test:SKIP_REGULAR_EXPRESSION), but I personally found this way a little bit more tricky.

Reply:
Thanks for the explanation, I am happy to go ahead with this